whole-repo-tests/test_type_hints.py

```diff
@@ -0,0 +1,71 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2018 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import os
+import subprocess
+
+import pytest
+
+import hypothesistooling as tools
+from hypothesistooling.scripts import pip_tool, tool_path
+
+
+def test_mypy_passes_on_hypothesis():
+    pip_tool('mypy', tools.PYTHON_SRC)
+
+
+def get_mypy_analysed_type(fname, val):
+    out = subprocess.Popen(
+        [tool_path('mypy'), fname],
+        stdout=subprocess.PIPE, encoding='utf-8', universal_newlines=True,
+        # We set the MYPYPATH explicitly, because PEP561 discovery wasn't
+        # working in CI as of mypy==0.600 - hopefully a temporary workaround.
+        env=dict(os.environ, MYPYPATH=tools.PYTHON_SRC),
+    ).stdout.read()
+    assert len(out.splitlines()) == 1
+    typ = out.split('error: Revealed type is ')[1].strip().strip("'")
```

Comment:
I have no idea what Mypy successful or unsuccessful output looks like – a brief comment here with examples of both would help satisfy me that this string parsing code is working correctly.

Reply:
I'll add a link to [the docs](https://mypy.readthedocs.io/en/latest/common_issues.html#displaying-the-type-of-an-expression) and an example of the output we're parsing.